Emby.Server.Implementations/Library/LibraryManager.cs

```diff
@@ -1710,13 +1712,20 @@ private async Task<IEnumerable<IntroInfo>> GetIntros(IIntroProvider provider, Ba
         /// <inheritdoc />
         public IEnumerable<BaseItem> Sort(IEnumerable<BaseItem> items, User? user, IEnumerable<ItemSortBy> sortBy, SortOrder sortOrder)
         {
-            var isFirst = true;
-
             IOrderedEnumerable<BaseItem>? orderedItems = null;
 
             foreach (var orderBy in sortBy.Select(o => GetComparer(o, user)).Where(c => c is not null))
             {
-                if (isFirst)
+                if (orderBy?.GetType() == typeof(RandomComparer))
```

Comment:
```suggestion
                if (orderBy is RandomComparer)
```

Untested but should work